### PR TITLE
refactor(samples): idtoken-cloudrun => idtoken-serverless

### LINKED ISSUE #1019: refactor(samples): Consolidate idtoken samples for serverless
## Background

Serverless products in general need two samples for Node.js: minting an identity token to make a request to a service or HTTP function, and validating an identity token on an arbitrary hosting platform receiving an HTTP request from Cloud Tasks or Cloud Scheduler.

## Proposal

1. Rewrite the [idtoken minting sample](https://github.com/googleapis/google-auth-library-nodejs/blob/master/samples/idtokens-cloudrun.js) `google_auth_idtoken_cloudrun` as `google_auth_idtoken_serverless`.
    * Remove Cloud Run from comments or replace with "Cloud Run or Cloud Functions"
    * Simplify the example URL to `https://RUN_OR_FUNCTION_TARGET_URL`
    * Remove "Cloud Run" from info log

2. Wrap in additional region tags for `run_service_to_service_auth` and `functions_bearer_token` for tracking as part of serverless work.

3. Update the [Cloud Run samples index](https://github.com/GoogleCloudPlatform/cloud-run-samples#readme) to reference the auth sample here.

My understanding is @bcoe is fine with this change. @ace-n could you confirm this approach for Cloud Functions?

### PR DESCRIPTION
**Pull request by grayside**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-auth-library-nodejs/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #1019 🦕

I did not see instructions on how to run the sample tests, so going to see what CI has to say.

### CODE REVIEW

**Comment by grayside** [comment]
Not clear how to run the lint/fix locally, and from this result I cannot tell what change is needed:

```
##[error]  41:18  error  Replace `⏎······`request·${url}·with·target·audience·${targetAudience}`⏎····` with ``request·${url}·with·target·audience·${targetAudience}``  prettier/prettier

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

I see the docs test failure, but I don't see other references within this repo and thus do not know where I need to make other changes:
```
[404] https://github.com/googleapis/google-auth-library-nodejs/blob/master/samples/idtokens-serverless.js


http://localhost:5758/
  [404] https://github.com/googleapis/google-auth-library-nodejs/blob/master/samples/idtokens-serverless.js
ERROR: Detected 1 broken links. Scanned 51 links in 1.649 seconds.
```

File: README.md
Comment (JustinBeckwith) [inline README.md:365]: Just double checking - have you done a check on devrel/snippets to see if we need to update paths anywhere?

File: samples/idtokens-serverless.js
Comment (JustinBeckwith) [inline samples/idtokens-serverless.js:40]: Looks like you need to run `npm run fix` and re-push - should help get the linter happy :)

**Review by JustinBeckwith — APPROVED** [review]
LGTM, after linting issues are resolved

**Comment by JustinBeckwith** [comment]
@grayside for the sample tests, `npm run samples-test` should get you where you want to go :)
https://github.com/googleapis/google-auth-library-nodejs/blob/master/CONTRIBUTING.md#running-the-tests

If that doesn't work, let us know we'll update the guide!

**Comment by grayside** [comment]
Ah, when I ran `npm run fix` I saw no output about fixes made and assumed nothing was done, but I see a change!

For `npm run samples-test`, I don't have access to the project used in this repo and don't see instructions on all the GCP resources that need to be setup. I'll fix the lint and we'll see what happens.

File: README.md
Comment (grayside) [reply README.md:365]: Does not appear in devrel/snippets.

**Comment by grayside** [comment]
Sample test was failing then went to passing. I assume there's some magic afoot...

**Comment by bcoe** [comment]
@grayside there's a chicken or the egg issue, where the link checker expects your sample to have already been landed:

```
  [404] https://github.com/googleapis/google-auth-library-nodejs/blob/master/samples/idtokens-serverless.js
```

If you add an ignore rule to `linkinator.config.json` for the above URL, we can then immediately revert that change once we land your work.

**Comment by JustinBeckwith** [comment]
If you're ok with this to be merged, I can also just hammer merge it.

**Comment by grayside** [comment]
That works for me. Btw, not sure how it works, but shouldn't I be the assignee? I have no rights here.

**Pull request merged**
